### PR TITLE
Add --debug flag to node.js process for dev task

### PR DESCRIPTION
**Pull request by perjg**
Add --debug flag for dev mode to node.js process (to allow attaching to process for debugging).

gulp-develop-server had to be updated to 0.4.3 to fix this issue:
https://github.com/narirou/gulp-develop-server/issues/10

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
     "gulp-cached": "1.x.x",
     "gulp-concat": "2.x.x",
     "gulp-declare": "^0.3.0",
-    "gulp-develop-server": "^0.2.5",
+    "gulp-develop-server": "^0.4.3",
```
Comment (rogatty) [inline package.json]: Please take some time to test if this won't start to crash server again. See https://github.com/Wikia/mercury/pull/809

Comment (perjg) [inline package.json]: Ok, I will run it for a while locally and then report back.

**Comment by perjg** [comment]
I've been running with 0.4.3 of gulp-develop-server for a while now, and I haven't experienced any issues. If this patch is interesting to have on master, someone else might want to try as well.

**Comment by rogatty** [comment]
After trying out `--debug` flag I've got spammed by messages:

```
[10:01:46] [INFO] Template changed: /home/igor/workspace/mercury/front/templates/main/main-page.hbs
[10:01:46] Starting 'scripts-front'...
[10:01:46] Starting 'templates'...
[10:01:46] Finished 'templates' after 535 ms
[10:01:47] [INFO] File changed: /home/igor/workspace/mercury/www/front/templates/main.js Restarting server
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
Failed to open socket on port 5858, waiting 1000 ms before retrying
[10:01:47] Development server was stopped. (PID:1980)
debugger listening on port 5858
```

There was more later on.

**Comment by kenkouot** [comment]
Needs more work. Closing for now.

**Pull request closed without merging**